Xamarin.Forms.Core/IndicatorView.cs

```diff
@@ -0,0 +1,256 @@
+			=> ((IndicatorView)bindable).ResetItemsSource((IEnumerable)oldValue));
+
+		public static readonly BindableProperty IndicatorLayoutProperty = BindableProperty.Create(nameof(IndicatorLayout), typeof(Layout<View>), typeof(IndicatorView), null, propertyChanged: TemplateUtilities.OnContentChanged);
+
+		public IndicatorView()
+			=> IndicatorLayout = new StackLayout { Orientation = StackOrientation.Horizontal };
+
+		IList<View> Items => IndicatorLayout.Children;
+
+		public Layout<View> IndicatorLayout
+		{
+			get => (Layout<View>)GetValue(IndicatorLayoutProperty);
+			set => SetValue(IndicatorLayoutProperty, value);
+		}
+
+		public int Position
+		{
+			get => (int)GetValue(PositionProperty);
+			set => SetValue(PositionProperty, value);
+		}
+
+		public int Count
+		{
+			get => (int)GetValue(CountProperty);
+			set => SetValue(CountProperty, value);
+		}
+
+		public int MaximumVisibleCount
+		{
+			get => (int)GetValue(MaximumVisibleCountProperty);
+			set => SetValue(MaximumVisibleCountProperty, value);
+		}
+
+		public DataTemplate IndicatorTemplate
+		{
+			get => (DataTemplate)GetValue(IndicatorTemplateProperty);
+			set => SetValue(IndicatorTemplateProperty, value);
+		}
+
+		public bool HidesForSingleIndicator
+		{
+			get => (bool)GetValue(HidesForSingleIndicatorProperty);
+			set => SetValue(HidesForSingleIndicatorProperty, value);
+		}
+
+		public Color IndicatorColor
+		{
+			get => (Color)GetValue(IndicatorColorProperty);
+			set => SetValue(IndicatorColorProperty, value);
+		}
+
+		public Color SelectedIndicatorColor
+		{
+			get => (Color)GetValue(SelectedIndicatorColorProperty);
+			set => SetValue(SelectedIndicatorColorProperty, value);
+		}
+
+		public double IndicatorSize
+		{
+			get => (double)GetValue(IndicatorSizeProperty);
+			set => SetValue(IndicatorSizeProperty, value);
+		}
+
+		public IEnumerable ItemsSource
+		{
+			get => (IEnumerable)GetValue(ItemsSourceProperty);
+			set => SetValue(ItemsSourceProperty, value);
+		}
+
+		void ResetIndicatorStyles()
+		{
+			try
+			{
+				BatchBegin();
+				ResetIndicatorStylesNonBatch();
+			}
+			finally
+			{
+				BatchCommit();
+			}
+		}
+
+		void ResetIndicatorCount(int oldCount)
+		{
+			try
+			{
+				BatchBegin();
+				if (oldCount < 0)
+				{
+					oldCount = 0;
+				}
+
+				if (oldCount > Count)
+				{
+					RemoveRedundantIndicatorItems();
+					return;
+				}
+
+				AddExtraIndicatorItems();
+			}
+			finally
+			{
+				ResetIndicatorStylesNonBatch();
+				BatchCommit();
+			}
+		}
+
+		void ResetIndicatorStylesNonBatch()
+		{
+			foreach (var child in Items)
+			{
+				ApplyColor(child as View);
+			}
+
+			IndicatorLayout.IsVisible = Count > 1 || !HidesForSingleIndicator;
+		}
+
+		void ResetItemsSource(IEnumerable oldItemsSource)
+		{
+			if (oldItemsSource is INotifyCollectionChanged oldObservableCollection)
```

Comment:
I'd name this `collection` instead of `oldObservableCollection` or `observableCollection` down below. Technically, there are different types of collections that implement `INotifyCollectionChanged` without being named observable.

Reply:
how about **oldCollection**?

Reply:
Sorry, that's what I meant. Use `oldCollection` and `collection` respectively.

Reply:
done